src/transformStateWithClones.js

```diff
@@ -6,8 +6,33 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  let modifiedState = { ...state };
+  const result = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        modifiedState = { ...modifiedState, ...action.extraData };
+        result.push({ ...modifiedState });
+        break;
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete modifiedState[key];
+        }
+        result.push(modifiedState);
+        break;
+      case 'clear':
+        modifiedState = {};
+        result.push(modifiedState);
```

Comment:
Similar to the previous comment, push a copy of `modifiedState` to ensure that subsequent modifications do not affect the already stored states.